dids/diddht/internal/bep44/bep44.go

```diff
@@ -25,77 +23,56 @@ type Message struct {
 
 	// The sequence number of the message, used to ensure the latest version of the data is retrieved
 	// and updated. It's a monotonically increasing number.
-	seq int64
+	Seq int64
 
 	// The signature of the message, ensuring the authenticity and integrity of the data. It's
 	// computed over the bencoded sequence number and value.
 	sig []byte
 
 	// The actual data being stored or retrieved from the DHT network, typically encoded in a format
 	// suitable for DNS packet representation of a DID Document.
-	v []byte
+	V []byte
 }
 
 // Signer is a function that signs a given payload and returns the signature.
 type Signer func(payload []byte) ([]byte, error)
 
 // NewMessage bencodes the payload, signes it with the signer and creates a new BEP44 message with the given sequence number, public key.
 func NewMessage(dnsPayload []byte, seq int64, publicKeyBytes []byte, signer Signer) (*Message, error) {
-	bencoded := map[string]any{
```

Comment:
this is effectively removed because of 2 reasons:
1. bencoding maps is non deterministic because we can't guarantee the order of the fields, which means the signature check fails
2. the bencoding standard wraps the map into starting with `d` and ending in `e`. to respect bep44 requirements, we need to remove the first and last byte
3. we only and always will have 2 fields here and bencoding results in a .. string. so, might as well use a string template to create the payload for signing